nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/common/AbstractMQTTProcessor.java

```diff
@@ -288,13 +288,27 @@ public ValidationResult validate(String subject, String input, ValidationContext
 
     public static Properties transformSSLContextService(SSLContextService sslContextService){
         Properties properties = new Properties();
-        properties.setProperty("com.ibm.ssl.protocol", sslContextService.getSslAlgorithm());
-        properties.setProperty("com.ibm.ssl.keyStore", sslContextService.getKeyStoreFile());
-        properties.setProperty("com.ibm.ssl.keyStorePassword", sslContextService.getKeyStorePassword());
-        properties.setProperty("com.ibm.ssl.keyStoreType", sslContextService.getKeyStoreType());
-        properties.setProperty("com.ibm.ssl.trustStore", sslContextService.getTrustStoreFile());
-        properties.setProperty("com.ibm.ssl.trustStorePassword", sslContextService.getTrustStorePassword());
-        properties.setProperty("com.ibm.ssl.trustStoreType", sslContextService.getTrustStoreType());
+        if (sslContextService.getSslAlgorithm() != null) {
```

Comment:
This is a very straightforward change, but would it be worth adding a unit test to verify that it resolves the problem?  The MqttTestUtils class has a createSslProperties() method, so a similar method that sets only the trust store properties could be added to support the use case described in NIFI-7895.

Reply:
I added unit test, thanks for the review.

Reply:
The changes and unit test look good.